src/Kernel-Tests/BlockClosureTest.class.st

```diff
@@ -150,11 +150,15 @@ BlockClosureTest >> testNew [
 
 { #category : #tests }
 BlockClosureTest >> testNoArguments [
+	| block1 block2 |
+	"avoid compile error in GemStone"
+	block1 := [:arg | 1 + 2].
+	block2 := [:arg1 :arg2 | 1 + 2].
 	[10
-		timesRepeat: [:arg | 1 + 2]]
+		timesRepeat: block1]
```

Comment:
GemStone has a compiler optimization that recognizes literal blocks passed with the #timesRepeat: message. By placing the block in a variable, we hide the true nature from the compiler. The Pharo test runs the same and gets the expected error at runtime.

Reply:
we have the same optimisation, but it is turned off by default, we should turn it on, too